src/qz/printer/action/PrintHTML.java

```diff
@@ -43,16 +43,29 @@ public void parseData(JSONArray printData, PrintOptions options) throws JSONExce
 
                 PrintingUtilities.Format format = PrintingUtilities.Format.valueOf(data.optString("format", "FILE").toUpperCase());
 
-                double pageWidth = PrinterJob.getPrinterJob().getPageFormat(null).getWidth();
+                double pageZoom = pxlOpts.getDensity() / 72.0;
+                if (pageZoom <= 0) { pageZoom = 1; }
+
+                double pageWidth = 0;
+                double pageHeight = 0;
+
+                // web dimension use 96dpi (or equivalent dp/metric)
```

Comment:
This was discovered when I tried to print a 2.25in div on a 2.25in label and it came out with excessive whitespace. Viewing the resulting pixel size of the html element showed 96dpi, a little investigating lead to sources like this stating the same. But the first three paragraphs under the heading are the focus
>Absolute length units represents a __physical measurement and when the physical properties of the output medium are known, such as for print layout__. This is done by anchored one of the unit to a physical unit and to defined the other relatively to it. The anchor is done __differently for low-resolution devices, like screens, and high-resolution devices, like printers__.

>For low-dpi devices, the unit px represents the physical reference pixel and the others are defined relative to it. __Thus, 1in is defined as 96px__ which equals 72pt. The consequence of this definition is that on such devices, length described in inches (in), centimeters (cm), millimeters (mm) doesn't necessary match the length of the physical unit with the same name.

>For high-dpi devices, inches (in), centimeters (cm), millimeters (mm) are defined as their physical counterparts. Therefore the __px unit is defined relative to them (1/96 of 1 inch)__.

For low end devices, it's basically saying the unit on the device won't match a physical inch because of this 96dpi conversion.
You're also reading the px css description, which for someone laying out a page for printing they shouldn't use because they cannot be sure of the printer dpi in all cases. These changes allow them to just use the inch/cm css units and get that same amount on the physical medium.